Makefile

```diff
@@ -0,0 +1,68 @@
+VERSION := 0.16.5
+
+# install directory layout
+PREFIX ?= /usr/local
+INCLUDEDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+PCLIBDIR ?= $(LIBDIR)/pkgconfig
+
+# collect sources
+ifneq ($(AMALGAMATED),1)
+	SRC := $(wildcard lib/src/*.c)
+	# do not double-include amalgamation
+	SRC := $(filter-out lib/src/lib.c,$(SRC))
+else
+	# use amalgamated build
+	SRC := lib/src/lib.c
+endif
+OBJ := $(SRC:.c=.o)
+
+# define default flags, and override to append mandatory flags
+CFLAGS ?= -O3
+override CFLAGS += -std=gnu99 -fPIC -Ilib/src -Ilib/include
+LDFLAGS ?=
+override LDFLAGS +=
```

Comment:
Just curious - what is the effect of these two `LDFLAGS` lines (compared to not explicitly defining LDFLAGS at all)?

Reply:
Nothing at all, but they serve as documentation on how to do it properly if you ever do add something here.

As demonstrated by the CFLAGS handling, `FOO ?=` will set the default if you haven't specified it in the environment, and `override FOO +=` will append to FOO (the override keyword is needed to ensure this happens even the defaults were passed as `make FOO=...` rather than `FOO=... make`).

It could be removed and left to be inferred from the previous two lines, or if there are any LDFLAGS you'd like to use, they could be added here.

Reply:
I think I’d slightly prefer to leave these out then. Thanks for the explanation.

Reply:
Done.